color-migration.js

```diff
@@ -0,0 +1,132 @@
+/*
+  This script will update the styles.scss files for each app to use the new color variables.
+  The files generated are based upon _styles.scss in the root of the project.
+  This script will take the colors from the defaultConfig.json and instanceConfig.json files in each app's config directory.
+  The colors in the instanceConfig.json file will override the colors in the defaultConfig.json file.
+  It is entirely possible that this approach will be replaced in the future by other Angular updates.
+  */
+const fs = require('fs/promises');
+const tinycolor = require('tinycolor2');
+
+// Crawls apps directory
+const getConfigs = async () => {
+  // Open apps directory
+  let dir = null;
+  try {
+    dir = await fs.opendir('./apps');
+  } catch (error) {
+    console.log("Error opening apps directory");
+    return;
+  }
+
+  const configs = [];
+  // Read directory
+  let dirent = null;
+  while ((dirent = await dir.read()) !== null) {
+    // Skip .gitkeep and e2e
+    if (dirent.name === '.gitkeep') continue;
+    if (dirent.name.includes('e2e')) continue;
+
+    const config = {
+      name: dirent.name,
+      path: `./apps/${dirent.name}`,
+      colors: {}
+    }
+    try {
+      const file = await fs.readFile(`./apps/${dirent.name}/src/assets/config/defaultConfig.json`, 'utf-8');
+      const obj = JSON.parse(file);
+      config.colors = obj["theme"];
+      // Update configs with instanceConfig.json
+      const file2 = await fs.readFile(`./apps/${dirent.name}/src/assets/config/instanceConfig.json`, 'utf-8');
+      const obj2 = JSON.parse(file2);
+      // Update missing colors
+      for (let color in config.colors) {
+        if (config.colors[color] === null) {
```

Comment:
feel like this doesn't work as intended

Reply:
Yes, I suppose it does not work as is should

Reply:
After tinkering with the script, I found out that the colours from instanceConfig.json are not that necessary to be used by this script. The AppConfigService will update the instance with the colours from the instanceConfig.json. Nevertheless, I updated the script to work properly